spring-pulsar-sample-apps/sample-reactive/src/main/java/org.springframework.pulsar.example/ReactiveSpringPulsarBootApp.java

```diff
@@ -50,66 +48,91 @@ public static void main(String[] args) {
-	 * consumer.
+	 * Sends string messages with a reactive template and receives them with a simple
+	 * reactive listener.
 	 */
 	@Configuration(proxyBeanMethods = false)
-	static class ReactiveSenderWithReactiveConsumer implements ApplicationListener<ApplicationReadyEvent> {
+	static class ReactiveTemplateWithSimpleReactiveListener implements ApplicationListener<ApplicationReadyEvent> {
 
 		private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
 		@Autowired
-		private ReactivePulsarTemplate<Foo> reactivePulsarTemplate;
+		private ReactivePulsarTemplate<String> reactivePulsarTemplate;
 
-		// TODO remove this once the auto-config is available
-		@Bean
-		ReactivePulsarConsumerFactory<Foo> reactivePulsarConsumerFactory(ReactivePulsarClient reactivePulsarClient) {
-			MutableReactiveMessageConsumerSpec spec = new MutableReactiveMessageConsumerSpec();
-			spec.setTopicNames(Collections.singletonList("sample-reactive-topic1"));
-			spec.setSubscriptionName("sample-reactive-sub1");
-			spec.setConsumerName("sample-reactive-consumer1");
-			return new DefaultReactivePulsarConsumerFactory<>(reactivePulsarClient, spec);
+		@Override
+		public void onApplicationEvent(ApplicationReadyEvent event) {
+			this.reactivePulsarTemplate
+					.send("sample-reactive-topic1", Flux.range(0, 10).map((i) -> "sample-message-" + i)).subscribe();
 		}
 
-		@Bean
-		ApplicationRunner listenPojo(ReactivePulsarConsumerFactory<Foo> reactiveConsumerFactory) {
-			return args -> {
-				ReactiveMessageConsumer<Foo> messageConsumer = reactiveConsumerFactory
-						.createConsumer(Schema.JSON(Foo.class));
-				messageConsumer.consumeMany((messageFlux) -> messageFlux.map(MessageResult::acknowledgeAndReturn))
-						.take(Duration.ofSeconds(10)).subscribe((msg) -> this.logger.info("Received: {}", msg));
-			};
+		@ReactivePulsarListener(subscriptionName = "sample-reactive-sub1", topics = "sample-reactive-topic1",
+				consumerCustomizer = "subscriptionInitialPositionEarliest")
+		public Mono<Void> listenSimple(String msg) {
+			this.logger.info("Simple reactive listener received: {}", msg);
+			return Mono.empty();
 		}
 
+	}
+
+	/**
+	 * Sends POJO messages with a reactive template and receives them with a reactive
+	 * streaming listener.
+	 */
+	@Configuration(proxyBeanMethods = false)
+	static class ReactiveTemplateWithStreamingReactiveListener implements ApplicationListener<ApplicationReadyEvent> {
+
+		private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+		@Autowired
+		private ReactivePulsarTemplate<Foo> reactivePulsarTemplate;
+
 		@Override
 		public void onApplicationEvent(ApplicationReadyEvent event) {
 			this.reactivePulsarTemplate.setSchema(Schema.JSON(Foo.class));
 			this.reactivePulsarTemplate
-					.send("sample-reactive-topic1", Flux.range(0, 10).map((i) -> new Foo("Foo-" + i, "Bar-" + i)))
+					.send("sample-reactive-topic2", Flux.range(0, 10).map((i) -> new Foo("Foo-" + i, "Bar-" + i)))
 					.subscribe();
```

Comment:
I would use
```java
			Flux.range(0, 10).map((i) -> new Foo("Foo-" + i, "Bar-" + i))
					.as(messages -> this.reactivePulsarTemplate.send("sample-reactive-topic2", messages)).subscribe();
```
Can't do suggestion because of deleted line.